src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -1555,21 +1556,21 @@ public bool IsExecutingInvocation(string invocationId)
             return _executingInvocations.ContainsKey(invocationId);
         }
 
-        public bool TryFailExecutions(Exception workerException)
+        public void Shutdown(Exception workerException)
         {
-            if (workerException == null)
-            {
-                return false;
-            }
+            WorkerShutdownException shutdownException = workerException is not null
+                ? new WorkerShutdownException("Worker encountered a fatal error and is shutting down.", workerException)
+                : new WorkerShutdownException("Worker encountered a fatal error and is shutting down.");
 
             foreach (var invocation in _executingInvocations?.Values)
             {
                 string invocationId = invocation.Context?.ExecutionContext?.InvocationId.ToString();
                 _workerChannelLogger.LogDebug("Worker '{workerId}' encountered a fatal error. Failing invocation: '{invocationId}'", _workerId, invocationId);
-                invocation.Context?.ResultSource?.TrySetException(workerException);
+
+                invocation.Context?.ResultSource?.TrySetException(shutdownException);
```

Comment:
Since you only new up this exception and never throw it, it won't have any stack trace on it. Is that acceptable? Or should you throw and catch it before calling `TrySetException`?